armory/cli/tools/collect_outputs.py

```diff
@@ -0,0 +1,581 @@
+            os.makedirs(output_path.with_suffix(""), exist_ok=True)
+            _safe_symlink(
+                result_path.parent,
+                output_path.with_suffix("") / result_path.parent.name,
+            )
+
+
+def collect_armory_outputs(command_args, prog, description):
+    """Collect results from armory output_directory and organize into tables."""
+    # We need to check both output dirs since `--no-docker` isn't passed
+    output_dir = _get_output_dir()
+    parser = argparse.ArgumentParser(
+        prog=prog,
+        description=description,
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    _add_parser_args(parser, output_dir=output_dir)
+    args = parser.parse_args(command_args)
+    update_filters(args.log_level, args.debug)
+    if args.collate is not False and args.collate is None:
+        args.collate = "config.scenario.name"
+
+    for at in args.unify or []:
+        if at not in HEADERS:
+            raise ValueError(f"Unrecognized attack {at}")
+
+    if args.default not in HEADERS:
+        raise ValueError(f"Unrecognized default attack {args.default}")
+
+    if args.clean:
+        return _clean(output_dir)
+
+    # get json results files
+    log.info(f"Recursive globbing for {args.glob} in {output_dir}")
+    results = list(
+        filter(
+            lambda p: p.parent.name != "saved_samples" and p.resolve() == p,
+            Path(output_dir).rglob("*/" + args.glob),
+        )
+    )
+    log.info(f"Found {len(results)} results files.")
+
+    # parse them into tables
+    tables = defaultdict(list)
+    collation = defaultdict(set)
+    for result in results:
+        log.debug(f"Loading {result}")
+        # load json
+        with open(result, "r") as f:
+            json_data = json.load(f)
+        # parse json
+        attack_name = json_data["config"]["attack"]["name"]
+        row = list(
+            _parse_json_data(
+                HEADERS.get(attack_name, HEADERS[args.default]),
+                json_data,
+                result,
+                absolute=args.absolute,
+            )
+        )
+        if args.filter is not None:
+            if not re.search(args.filter, row[0]):
+                log.debug(f"Skipping {result} due to filter")
+                continue
+        if args.collate is not False:
+            try:
+                tgt_collate = reduce(
+                    operator.getitem, args.collate.split("."), json_data
+                )
+            except (KeyError, TypeError):
+                log.debug(f"Could not collate {result} on {args.collate}")
+            else:
+                collation[str(tgt_collate)].add(attack_name)
+        tables[attack_name].append(row)
+    log.info(f"Parsed {len(tables)} results.")
+
+    # collate results
+    if args.collate is not False:
+        for tgt_collate, attacks in collation.items():
+            # make sure all attacks have the same headers
+            first = attacks.pop()
```

Comment:
Pop removes the element, need to add it back!

Reply:
Addressed by https://github.com/twosixlabs/armory/pull/1986/commits/48f1efe5f74b2b266a450c2fc1a2be4c2b87705a